### DISCUSSION TITLE
刚刚部署的时候能用，过一段时间后就提示错误

### DISCUSSION THREAD

**Discussion by QJL7** [spencerwooo/onedrive-vercel-index#287] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/37794963/149897893-98adb832-10a5-432c-841e-69000bc38693.png)
刚刚部署的时候能用，过一段时间后就提示错误。
这个问题怎解决

**Comment by spencerwooo** [comment, marked as answer]
这还是第一次见这么长的报错……本地能复现吗？

**Reply by QJL7** [reply]
看了下错误报告，好像是认证不通过的原因。可是我的E5账号是自己的管理员账号，应该没问题才对的吧。
是需要把clientId和obfuscatedClientSecret换成我的吗？
![image](https://user-images.githubusercontent.com/37794963/149907139-796cf43d-a489-48b2-aaee-a72b876c24cd.png)

**Reply by spencerwooo** [reply]
可以换的：https://docs.spencerwoo.com/using-your-own-client-id-and-secret

有部分教程还没写完，但是大致流程就是这样的

**Reply by spencerwooo** [reply]
如果你认证成功了，token 都保存到 Redis 了，应该没啥问题理论上…但是这个是 runtime 错误，也不太好 debug

**Reply by QJL7** [reply]
现在就是刚刚部署的时候认证成功token也保存到Redis了，过了一段时间后就不行了。。。

**Reply by spencerwooo** [reply]
不过它说 `invalid client secret`，可能是跟你 E5 账号有关系？试试用自己的 client id 和 client secret 吧我建议

**Reply by QJL7** [reply]
已经改上， 过一段时间看看，谢谢了

**Reply by QJL7** [reply]
我发现redis的access_token过期后就没有重新获取了，所以就出现了认证失败的问题，不知道怎么解决

**Reply by spencerwooo** [reply]
不应该啊，后面我再本地排查一下吧，但是如果是 access_token 不更新，不应该只有你一个 E5 的版本失效啊，理论上所有版本都可能失效，难道还是我现在更新那部分写的有问题🤔

**Reply by QJL7** [reply]
解决了，我把Node.js版本换成12就行了...

**Reply by spencerwooo** [reply]
啊？这……好吧

**Reply by aloeJun** [reply]
我也出现了同样问题，修改了Node版本还是不可以，是否还有其它操作？

**Reply by QJL7** [reply]
api.json不用改，用原来的。还有重新创建了一次redis

**Comment by Ovler-Young** [comment]
试试看检查一下redis状态，昨天upslash跑超了，报错和这个一样。
